Sanaz filtering backend

SanazNajiyan opened the pull request.

SanazNajiyan committed:
Fix linting

SanazNajiyan committed:
fix merge conflicts

SanazNajiyan committed:
Merge branch 'dev' of https://github.com/TheMinefreak23/fairreckitlib into sanaz-countFiltering-backend

SanazNajiyan committed:
Merge branch 'dev' into sanaz-countFiltering-backend

SanazNajiyan committed:
count filter

SanazNajiyan committed:
Merge branch 'dev' of https://github.com/TheMinefreak23/fairreckitlib into sanaz-countFiltering-backend

SanazNajiyan committed:
generalizing the gender filter

SanazNajiyan committed:
add filter count+ test

SanazNajiyan committed:
improve the code

SanazNajiyan committed:
filter generalizing and improvement

LCYC1 reviewed (changes requested):
hi, i'm afraid there was a miscommunication on the count filter's functionality. It must be as follows:

_input_: count (int), dataframe with column country (or some other categorical column).
_body_: Filter out all countries with less than count entities in the dataframe.
_output_: dataframe

**example1**
_input_: 2, dataframe with country column: ["NL", "NL", "NL", "BE", "NL", "BE", "FR"]
_output_: dataframe with country column: ["NL", "NL", "NL", "BE", "NL", "BE"]

**example2**
_input_: 3, dataframe with country column: ["NL", "NL", "NL", "BE", "NL", "BE", "FR"]
_output_: dataframe with country column: ["NL", "NL", "NL", "NL"]

**example3**
_input_: 3, dataframe with random chinese column: ["你好!", "你好!", "奶黄包", "你好!", "再见!", "再见!"]
_output_: dataframe with random chinese column: ["你好!", "你好!", "你好!"]

**example4**
_input_: 1, dataframe with random chinese column: ["你好!", "你好!", "奶黄包", "你好!", "再见!", "再见!"]
_output_: dataframe with random chinese column: ["你好!", "你好!", "奶黄包", "你好!", "再见!", "再见!"]

**example5**
_input_: 0, dataframe with random dutch column: ["hoi", "hoi", "doei", "goeiemiddag", "moguh"]
_output_: dataframe with random dutch column: []

LCYC1 commented:
The unit tests are fine, they could be tested on other filters.

SanazNajiyan committed:
refactoring filters based on range or list of values

SanazNajiyan committed:
Fixing init docstrings

SanazNajiyan committed:
fixing docstrings

SanazNajiyan committed:
fix docstring

SanazNajiyan committed:
Merge branch 'dev' of https://github.com/TheMinefreak23/fairreckitlib into sanaz-countFiltering-backend

SanazNajiyan committed:
fixing docstring

SanazNajiyan committed:
FilterCountAttrib + unit test

nput: count (int), dataframe with column country (or some other categorical column).
body: Filter out all countries with less than count entities in the dataframe.
output: dataframe

example1
input: 2, dataframe with country column: ["NL", "NL", "NL", "BE", "NL", "BE", "FR"]
output: dataframe with country column: ["NL", "NL", "NL", "BE", "NL", "BE"]

example2
input: 3, dataframe with country column: ["NL", "NL", "NL", "BE", "NL", "BE", "FR"]
output: dataframe with country column: ["NL", "NL", "NL", "NL"]

LCYC1 commented:
gottem and your last commit!
